keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/unary/TADVertex.java

```diff
@@ -0,0 +1,39 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.unary;
+
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.diff.DualNumber;
+
+import java.util.Map;
+
+import static io.improbable.keanu.tensor.TensorShapeValidation.shapeAlongDimension;
+
+public class TADVertex extends DoubleUnaryOpVertex {
```

Comment:
This should be called `DoubleTADVertex` for consistency with the Integer and Boolean versions.

Reply:
Our current convention is actually to leave the Double type blank and only fill it in for Integer and Bool. We should have a chat about this!